app/vue/src/client/preview/render.ts

```diff
@@ -44,10 +44,8 @@ export default function render({
 
   showMain();
 
-  // at component creation || refresh by HMR
-  if (!root[COMPONENT] || !forceRender) {
-    root[COMPONENT] = element;
-  }
```

Comment:
I don't really understand what this line does but this is inconsistent with the other frameworks, AFAIK.

In React, the *only* time the component is not re-mounted is when `forceRender` is true. So the behaviour is approximately:

1. If you change the implementation (HMR), remount,
2. If you change story (on the same component), remount,
3. If you change knobs/controls, **DO NOT** remount.

In https://github.com/storybookjs/storybook/issues/10616 @rhendric and I discuss case 1. above in detail (it would be nicer if it didn't remount). 

I can see arguments both ways for case 2. but people seem OK with the current behaviour.

However, it seems like changing 3 would be bad. If your story includes an iframe or some other element that doesn't like to be remounted this would mean you have a bad time using controls.

Reply:
agree with @tmeasday . With this fix, we observe that CSS transitions aren't rendered on props changes. DOM node is remounted so there is no animation. Would be great if we could have the same Vue component with new set of properties.

Reply:
@kvas-damien if we remount, css animation are restarted, if we do NOT remount css transitions work.

Reply:
Thanks guys, I've re-opened the issue, and will try a different fix (hopefully today)